seguro/common/test_store.py

```diff
@@ -21,27 +21,73 @@ def test_store():
     f.close()
 
     # Put file into storage
-    storage.put_file("myStorageFile.txt", "myfile.txt")
-
-    # Check if file has changed...
-    # Note: As the storage client cannot know if the file changed on
-    #       the first call file_changed() will always return True on first call
-    assert storage.file_changed("myStorageFile.txt") is True
-    assert storage.file_changed("myStorageFile.txt") is False
+    store.put_file("myStorageFile.txt", "myfile.txt")
 
     # Write new content to file
-    # FIXME: last_modified attribute of S3Storage only retruns
-    #        second resolution...
-    time.sleep(1)
-    storage.write_to_file("myStorageFile.txt", "!egarotS3S olleH")
+    store.put_file_contents("myStorageFile.txt", "!egarotS3S olleH")
 
-    # If file has changed (which it should at this point), download it...
-    if storage.file_changed("myStorageFile.txt"):
-        print("Info: myStorageFile.txt has changed - downloading it again...")
-        storage.get_file("myLocalStorageFile.txt", "myStorageFile.txt")
+    store.get_file("myLocalStorageFile.txt", "myStorageFile.txt")
 
     f = open("myLocalStorageFile.txt", "r")
     content = f.read()
 
     assert content == "!egarotS3S olleH"
     print(content)
+
+
+def test_watch():
+    store = Client()
+
+    filename = "some/prefix/test"
+
+    store.remove_file(filename)
+
+    def create_obj():
+        store.put_file_contents(filename, "this is a test2")
+        store.remove_file(filename)
+
+    t = threading.Timer(0.1, create_obj)
+    t.start()
+
+    events = store.watch("some/prefix/")
+
+    event = next(events)
+    assert event.type == EventType.CREATED
+    assert event.filename == filename
+
+    event = next(events)
+    assert event.type == EventType.REMOVED
+    assert event.filename == filename
+
+
+def test_watch_async():
+    store = Client()
+
+    i = 0
+
+    def callback(event: Event):
+        nonlocal i
+
+        if i == 0:
+            assert event.type == EventType.CREATED
+            assert event.filename == filename
+        elif i == 1:
+            assert event.type == EventType.REMOVED
+            assert event.filename == filename
+
+            # Raise a StopIteration exception to stop processing events
+            raise StopIteration()
+
+        i += 1
+
+    filename = "some/prefix/test"
+
+    store.remove_file(filename)
+    watcher = store.watch_async("some/prefix/", callback)
+
+    watcher.start()

```

Comment:
Is there a reason to force the user call `watcher.start()` manually after defining which prefix to watch?
Couldn't that be simplified by starting the watcher as soon as a prefix is registered for async watching, and stopping it when the watcher object is destroyed?

Reply:
Yeah, thats a good point. We should start the watcher right away :)

Reply:
I was following the sematics of `threading.Thread` which also needs to be started manually

Reply:
I see. If possible I would still go with starting it right away... :)

Reply:
Addressed in 8404727468f01b3227682f0363ad6637ab1810c8